src/UI/Actions.hs

```diff
@@ -237,10 +331,9 @@ updateMailInList m s =
 setError :: Error -> AppState -> AppState
 setError = set asError . Just
 
-reloadListOfMails :: Vector NotmuchMail -> AppState -> AppState
-reloadListOfMails vec =
-  set (asMailIndex . miListOfMails) (L.list ListOfMails vec 1)
-  . set asAppMode BrowseMail
+updateMailsList :: Vector NotmuchMail -> AppState -> AppState
```

Comment:
I think this is buggy. If you carry out a search which yields no results and an additional search on top of it which yields results, the current selected index is Nothing.